lib/modules/versioning/unity3d/index.ts

```diff
@@ -0,0 +1,81 @@
+import semver from 'semver';
+import { regEx } from '../../../util/regex';
+import { GenericVersion, GenericVersioningApi } from '../generic';
+import type { VersioningApi } from '../types';
+
+export const id = 'unity3d';
+export const displayName = 'Unity3D';
+export const urls = [
+  'https://docs.unity3d.com/Manual/ScriptCompilationAssemblyDefinitionFiles.html#version-define-expressions:~:text=characters%20are%20supported.-,Unity%20version%20numbers,-Current%20versions%20of',
+];
+export const supportsRanges = false;
+
+class Unity3dVersioningApi extends GenericVersioningApi {
+  private static readonly ReleaseStreamType = new Map([
+    ['Alpha', 'a'],
+    ['Beta', 'b'],
+    ['Patch', 'p'],
+    ['Experimental', 'x'],
+    ['Stable', 'f'],
+    ['Stable (China)', 'c'],
+  ]);
+  private static readonly ReleaseStreamTypeKeyOrder = Array.from(
+    Unity3dVersioningApi.ReleaseStreamType.values(),
+  );
+
+  protected _parse(version: string): GenericVersion | null {
+    const matches = regEx(
+      /^(?<Major>\d+)\.(?<Minor>\d+)\.(?<Patch>\d+)(?<ReleaseStream>\w)(?<Build>\d+)/gm,
+    ).exec(version);
+    if (!matches?.groups) {
+      return null;
+    }
+    const { Major, Minor, Patch, ReleaseStream, Build } = matches.groups;
+
+    const release = [
+      parseInt(Major),
+      parseInt(Minor),
+      parseInt(Patch) * 1000 + parseInt(Build),
```

Comment:
why combining this? 
```suggestion
      parseInt(Patch),
      parseInt(Build),
```
I would prefer to compare them separately which is already support by base class

Reply:
Not sure why I was under the impression the GenericVersioningApi comparison was limited to three entries...

I've reworked it to only use generic versioning inside fac18e522406962e7150c39c40445d7e3f33b4bd; this achieves the same functionality and saves a lot of code. Thanks for pointing it out!